### PR TITLE
Fix timing allow check

### LINKED ISSUE #24367: Timing allow check should only run for cross-origin requests
Right now we perform the check for every response we get and don't actually compare the current request's origin (or any of the requests in its redirect chain) against the origin of the original source of the request. This means that same-origin requests are treated as failing the timing allow check, which is incorrect.

### PR DESCRIPTION
**Pull request by shnmorimoto**
Fix #24367

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #24367 (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because ___

### CODE REVIEW

File: components/net/http_loader.rs [hunk 1]
```diff
@@ -1462,7 +1462,15 @@ fn http_network_fetch(
             _ => false,
         });
 
-    if !req_origin_in_timing_allow && !wildcard_present {
+    let is_same_origin = request.url_list.iter().any(|url| match request.origin {
```
Comment (jdm) [inline components/net/http_loader.rs]: We should either call this `is_cross_origin` and check if any of the urls in the url list are not same-origin, or we should call it `is_same_origin` and check if _all_ of the urls in the url list are same-origin.

File: components/net/http_loader.rs [hunk 2]
```diff
@@ -1462,7 +1462,15 @@ fn http_network_fetch(
...
+        SpecificOrigin(ref immutable_request_origin) => {
+            url.clone().into_url().origin().ascii_serialization() ==
+                immutable_request_origin.ascii_serialization()
```
Comment (jdm) [inline components/net/http_loader.rs]: Let's do `url.origin() == immutable_request_origin` or `url.origin() != immutable_request_origin` instead.

**Review by jdm — CHANGES_REQUESTED** [review]
Looks good! I'll do a test run once the changes to perform the same-origin check for all URLs are present.

File: components/net/http_loader.rs (hunk 1 above)
Comment (shnmorimoto) [reply components/net/http_loader.rs]: Thanks for the review.

I've fixed it as following.

> check if all of the urls in the url list are same-origin.

and

```rust
let is_same_origin = request.url_list.iter().all(|url| match request.origin {
```

File: components/net/http_loader.rs (hunk 2 above)
Comment (shnmorimoto) [reply components/net/http_loader.rs]: Thanks.
I've fixed it.

```rust
url.origin() == *immutable_request_origin
```

**Comment by jdm** [comment]
@bors-servo r+
Thanks!

**Comment by jdm** [comment]
@bors-servo retry
* https://github.com/rust-lang/rust/issues/65536

**Pull request merged**